### PR TITLE
Move `internal/plugin` to `pkg`

### PR DESCRIPTION
**Pull request by pkosiec**
## Description

Changes proposed in this pull request:

- Move `internal/plugin` to `pkg`

**Reason:** The plugin index and other types should be exported to be used externally.

### CODE REVIEW

File: pkg/plugin/testdata/TestGetFirstFileInDirectory/hidden_and_exe_files/._DS_Store
Comment (vaijab) [inline pkg/plugin/testdata/TestGetFirstFileInDirectory/hidden_and_exe_files/._DS_Store:1]: This is probably not needed?

**Review by vaijab — APPROVED** [review]
LGTM. Many files.. :-)

File: pkg/plugin/testdata/TestGetFirstFileInDirectory/hidden_and_exe_files/._DS_Store
Comment (pkosiec) [reply pkg/plugin/testdata/TestGetFirstFileInDirectory/hidden_and_exe_files/._DS_Store:1]: Actually it is, as it is used for the downloader unit tests: https://github.com/kubeshop/botkube/blob/9ac149cc8cfc14f66cc1c64e9e23934ed42a9652/pkg/plugin/downloader_test.go

We want to make sure such files are ignored - specifically, this condition: https://github.com/kubeshop/botkube/blob/6ad2cd03c2171cfecc29ca2e40a87ce56d769306/pkg/plugin/downloader.go#L115

**Comment by pkosiec** [comment]
Merging as the teams + slack tests seems to be flaky 🤔 
```
     bots_test.go:254: Creating Botkube Cloud instance...
    bots_test.go:1772: 
        	Error Trace:	/home/runner/work/botkube/botkube/test/e2e/gql_client.go:818
        	            				/home/runner/work/botkube/botkube/test/e2e/bots_test.go:1772
        	            				/home/runner/work/botkube/botkube/test/e2e/bots_test.go:257
        	            				/home/runner/work/botkube/botkube/test/e2e/bots_test.go:174
        	Error:      	Received unexpected error:
        	            	Message: 504 Gateway Timeout; body: "<html>\r\n<head><title>504 Gateway Time-out</title></head>\r\n<body>\r\n<center><h1>504 Gateway Time-out</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n", Locations: [], Extensions: map[code:request_error]
        	Test:       	TestTeams
```
I'll take a look at this problem later today.

**Pull request merged**